src/lib.rs

```diff
@@ -604,6 +604,13 @@ impl Contract {
             "This account is only allowed to change proposal status from DRAFT to REVIEW"
         );
 
+        if self.has_moderator(editor_id.clone()) && !proposal_body.supervisor.is_some() {
+            require!(
+                current_timeline == proposal_body.timeline,
+                "As a moderator, you can't change the timeline status of the proposal without adding a supervisor"
+            );
+        }
```

Comment:
1. We require a supervisor for the Approval/ConditionalApproval and later stages OR, in other words, supervisor is optional while the proposal is still in draft or review - please, come up with a better check.
2. It does not make sense to only check this rule for moderators (it feels like someone else can bypass the rule, which should not be the case!)